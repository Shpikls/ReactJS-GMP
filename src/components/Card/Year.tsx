import * as React from 'react'
import styled from 'styled-components'

type PropsYear = {
	year: string
}

export const Year = ({ year }: PropsYear): JSX.Element => {
	return <YearStyled>{year}</YearStyled>
}

const YearStyled = styled.div`
	position: absolute;
	top: 0;
	right: 0;
	font-family: Montserrat, sans-serif;
	font-style: normal;
	font-weight: 500;
	font-size: 14px;
	line-height: 17px;
	text-align: center;
	color: #ffffff;
	opacity: 0.7;
	padding: 4px 12px;
	border: 1px solid #979797;
	box-sizing: border-box;
	border-radius: 4px;
`
