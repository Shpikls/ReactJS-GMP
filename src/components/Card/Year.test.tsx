import { render } from '@testing-library/react'
import React from 'react'
import { Year } from './Year'

describe('Testing Year.tsx', () => {
	test('To match snapshot', () => {
		const { asFragment } = render(<Year year="9889" />)

		expect(asFragment()).toMatchSnapshot()
	})

	test('To null returned', () => {
		const { container } = render(<Year />)
		const div = container.querySelector('div')
		expect(div).toBe(null)
	})
})
