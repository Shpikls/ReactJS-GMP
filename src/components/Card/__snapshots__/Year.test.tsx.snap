// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Year.tsx To match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBQB jQDGOs"
  >
    9889
  </div>
</DocumentFragment>
`;

exports[`Testing Year.tsx To match snapshot 2`] = `<DocumentFragment />`;
